src/components/calculator/SavedDataSelector.tsx

```diff
@@ -42,20 +61,167 @@ const SavedDataLink: React.FunctionComponent<{
     </Button>
   )
 }
+
+// show [Build your own scenario] if no preset found
+const SavedScenarioContinue: React.FunctionComponent<{
+  variant: string
+  scenarioName: string
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+}> = (props): React.ReactElement => {
+  const { t } = useTranslation()
+  return (
+    <div className="mt-2">
+      {t('calculator.no_prebuilt_scenario_found', {
+        scenarioName: props.scenarioName,
+      })}
+      <Button
+        className="mt-1 d-block"
+        variant="primary"
+        onClick={props.onClick}
+      >
+        {t('calculator.saved_scenario_continue_custom')}
+      </Button>
+    </div>
+  )
+}
+
+const InteractionTypeIcon: React.FunctionComponent<{
+  interaction: string
+}> = (props): React.ReactElement => {
+  const iconFromInteractionType = (interaction: string) => {
+    const iconProps = { size: 24 }
+    if (props.interaction === 'oneTime') {
```

Comment:
why is this props.interaction in some lines and interaction in others?

Also, consider a switch block instead?

Reply:
Done